section2/crud.rb

```diff
@@ -2,22 +2,26 @@ module Crud
   require 'bcrypt'
   puts "Module CRUD activated"
 
-  def self.create_hash_digest(password)
+  # def self.create_hash_digest(password)
+  def create_hash_digest(password)
     BCrypt::Password.create(password)
   end
 
-  def self.verify_hash_digest(password)
+  # def self.verify_hash_digest(password)
+  def verify_hash_digest(password)
     BCrypt::Password.new(password)
   end
 
-  def self.create_secure_users(list_of_users)
+  # def self.create_secure_users(list_of_users)
+  def create_secure_users(list_of_users)
```

Comment:
It don't make sense for this to be without `self.` At least to me. Can you explain why you remove the `self.` ?

Reply:
When using the `self.` , it makes the method to become a class method. `self.` is removed to demonstrate the use of `include` which the method will have to be instance method.

Reply:
i know what self and include does. 
You still didn't answer the question.
THink. Why I single this method out of all the others? Whats so special about this?

Reply:
> i know what self and include does.
> You still didn't answer the question.


`self.` is removed to make it an instance method. As this method can only be called when a class has been instantiate. 

> THink. Why I single this method out of all the others? Whats so special about this?

From what I see, this method is calling another method in the module inside. 

If this method consist of `self.` , wouldn't the another method inside here: `create_hash_digest` (with no `self.`) result in error? If `self.` is added for this method & calling it with `Crud.create_secure_users(list_of_users)`, this would result in error since `create_hash_digest` method is an instance method.

Reply:
I asked why. You never answer why. You just said what.

And from your case, it all depends on how you structure your code

Reply:
> THink. Why I single this method out of all the others? Whats so special about this?

Is it because `using_crud.rb` file is also using this method: `Crud.create_secure_users(list_of_users)` , and if I removed the `self.` here, it won't work in that file.

Reply:
No. Anyway, the answer is, you should leave the `self` here. Because this is a method that create secure a list of users.
If you don't put the self, then what will happen is that what you are doing with your code is:

1 user secure every user including himself. Which don't make sense because other users have no relation with that 1 user.

eg:
```
student1.create_secure_users(full_students_list)
```

Why is creating secure users under student1? Don't make any sense at all.

Reply:
Ok, does that mean the changes for this method right here: -
```
def verify_hash_digest
  BCrypt::Password.new(self.password)
end
```
Would have to change to put the `self.verify_hash_digest` and the `self.password` instance method here wouldn't work anymore?

Or do I just write `BCrypt::Password.new(student_list[:password])` inside the `.create_secure_users(full_students_list)` method? Since `.create_secure_users(full_students_list)` is using `verify_hash_digest` method currently.

Reply:
verify_hash_digest should be without `self`
Think about it. How can you design this flow to make it dry